In [2]:
import pandas as pd

In [3]:
data_dir = "./data/"

Read in the data

In [5]:
cv_table = pd.read_csv(data_dir + "cv_table.csv")
ndr_table = pd.read_csv(data_dir + "ndr_table.csv")
poll_table = pd.read_csv(data_dir + "poll_table.csv")

latlong_cv = pd.read_csv(data_dir + "latlong_cv.csv")
latlong_ndr = pd.read_csv(data_dir + "latlong_ndr.csv")
latlong_pollination = pd.read_csv(data_dir + "latlong_pollination.csv")


We have three tables of data, one for each contribution of nature. We also have three tables relating each data point to a longitude and a latitude.

Let's look at the tables first!

# cv_table

In [66]:
cv_table.describe(include = 'all')

fid         UN_cur        UN_ssp1        UN_ssp3  \
count  686665.000000  686665.000000  686665.000000  686665.000000   
mean   343333.000000       2.482690       2.822749       2.982682   
std    198223.255633       0.511879       0.573237       0.618028   
min         1.000000       1.030057       1.237034       1.297790   
25%    171667.000000       2.082644       2.376177       2.492883   
50%    343333.000000       2.440013       2.777381       2.935599   
75%    514999.000000       2.817269       3.259844       3.419952   
max    686665.000000       4.291871       4.817462       5.000000   

             UN_ssp5        pop_cur       pop_ssp1       pop_ssp3  \
count  686665.000000  686665.000000  686665.000000  686665.000000   
mean        3.139066     485.270589     586.541642     605.606714   
std         0.644773    2772.848940    3590.280849    3368.830041   
min         1.311983       0.000000       0.000000       0.000000   
25%         2.667168       0.510938       0.423553       0.455772   
50%         3.107233      21.314615      21.012911      26.209813   
75%         3.619904     156.068106     153.478002     184.031578   
max         5.000000  236893.117288  270175.660403  247236.351206   

            pop_ssp5  
count  686665.000000  
mean      608.113528  
std      3719.613422  
min         0.000000  
25%         0.445830  
50%        21.459414  
75%       159.716874  
max    270048.141080

This table describes the coastal risk mitigation by natural habitats.
Some explanation of the table follow:

* **fid** - the id of the geographical location that the data describes. This id can be converted to lon/lat using the latlong table.  
* **UN_cur** - the **current** (2015) unmet need of the coastal risk mitigation in this particular area. If this figure is negative then it means that we have enough, and even a surplus, in this particular area. 
* **UN_ssp1** the unmet need of the coastal risk mitigation in the year 2050, given that scenario **ssp1** is realized. 
* **ssp1, ssp3, ssp5** - the three future scenarios. It is not perfectly clear which scenraio is which here. ssp1 is most likely the sustainability scenario, purely based on the observation that the unmet need in this column is lower than for the other scenarios. The other two scenarios are fossil-fuels and regional rivalry.
* **pop_cur** - the **current** (2015) number of people in this geographic region who are affected by coastal risk mitigation. If the unmet need is high and the population is high then a lot of people are negatively affected.
* **pop_sspX** - same as the above, but in the year 2050, for the three different future scenarios.

# ndr_table

In [65]:
ndr_table.describe(include = 'all')

fid     country   region        UN_cur       UN_ssp1  \
count   25243.000000       25243    25243  2.524300e+04  2.524300e+04   
unique           NaN         228        8           NaN           NaN   
top              NaN  Antarctica  UNKNOWN           NaN           NaN   
freq             NaN        6511     6557           NaN           NaN   
mean    31136.198669         NaN      NaN  2.847848e+06  2.982161e+06   
std     20508.712007         NaN      NaN  8.093390e+06  9.765498e+06   
min      2262.000000         NaN      NaN  0.000000e+00  0.000000e+00   
25%     13083.500000         NaN      NaN  0.000000e+00  0.000000e+00   
50%     25299.000000         NaN      NaN  2.304164e+03  7.343517e+02   
75%     56810.500000         NaN      NaN  2.418531e+06  1.757585e+06   
max     64799.000000         NaN      NaN  1.714816e+08  3.159408e+08   

             UN_ssp3       UN_ssp5     rurpopcur    rurpopssp1    rurpopssp3  \
count   2.524300e+04  2.524300e+04  2.524300e+04  2.524300e+04  2.524300e+04   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean    3.464451e+06  2.888546e+06  1.693007e+05  1.008505e+05  2.328594e+05   
std     1.072992e+07  8.570622e+06  6.919647e+05  4.071316e+05  9.817925e+05   
min     0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
25%     0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
50%     7.217091e+02  7.175778e+02  0.000000e+00  0.000000e+00  0.000000e+00   
75%     2.092982e+06  1.920021e+06  4.479987e+04  2.482487e+04  4.230556e+04   
max     2.093042e+08  2.064157e+08  1.667823e+07  8.933015e+06  2.162865e+07   

          rurpopssp5  
count   2.524300e+04  
unique           NaN  
top              NaN  
freq             NaN  
mean    1.003789e+05  
std     4.030947e+05  
min     0.000000e+00  
25%     0.000000e+00  
50%     0.000000e+00  
75%     2.592468e+04  
max     8.607843e+06

This table describes the improvement of water quality by filtering of pollutants (in our case probably excessive amounts of nutrients, making the water quality bad) when water passes by natural habitats such as forests and wetlands.

The columns are the same as in the previous table, although `pop` is replaced by `rurpop`. Charlotte has confirmed that this is still the same metric though.

We also have two extra columns `country` and `region`. We will most probably not use these, especially as we don't actually have them for each table.

One thing we can immediately observe is that the count (number of entries) is much lower, the previous table had 686665 entries while this one has only 25243. This needs to be investigated. Does this mean that we don't have data for the whole earth for this particular contribution of nature?

# poll_table

In [64]:
poll_table.describe(include = 'all')

fid        UN_cur       UN_ssp1       UN_ssp3       UN_ssp5  \
count   9160.000000  9.160000e+03  9.160000e+03  9.160000e+03  9.160000e+03   
mean   25270.656441  9.630618e+04  8.972863e+04  9.375806e+04  9.990251e+04   
std    10032.053267  4.348216e+05  4.136170e+05  4.113846e+05  4.335244e+05   
min     8845.000000  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
25%    16672.000000  2.088596e+00  2.046068e+00  9.315915e+00  9.550000e+00   
50%    22971.500000  1.027879e+03  9.781146e+02  1.492830e+03  1.594218e+03   
75%    33446.250000  3.107308e+04  2.827017e+04  3.439839e+04  3.474784e+04   
max    51951.000000  2.322062e+07  2.249534e+07  2.321814e+07  2.270223e+07   

            pop_cur      pop_ssp1      pop_ssp3      pop_ssp5  
count  9.160000e+03  9.160000e+03  9.160000e+03  9.160000e+03  
mean   2.574799e+05  3.993871e+05  4.659653e+05  3.603159e+05  
std    1.004224e+06  1.546353e+06  1.681148e+06  1.450151e+06  
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  
25%    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  
50%    6.223414e+02  3.940410e+03  3.460522e+03  0.000000e+00  
75%    1.263994e+05  1.874406e+05  2.276243e+05  1.468894e+05  
max    2.338777e+07  3.778232e+07  3.282686e+07  3.222564e+07

This table describes the pollination of crops coming from natural habitats inhabiting pollinators (ex. bees).

The columns are the same as previously.

Here though we have an even lower count. What does this mean? A hypothesis could be that the resolution is simply not just not as high for the data here.

# latlong_cv

In [63]:
latlong_cv.describe(include = 'all')

fid            lat            lng
count  686665.000000  686665.000000  686665.000000
mean   343333.000000       9.346826      23.706301
std    198223.255633      32.202816     102.750279
min         1.000000     -58.480000    -180.000000
25%    171667.000000     -10.600000     -73.070000
50%    343333.000000       9.600000      12.640000
75%    514999.000000      35.450000     123.250000
max    686665.000000      62.800000     180.000000

This looks good, just a simple mapping from the **fid** to the coordinates. We have the same number of mappings as we have entrys in the `cv_table`.

# latlong_ndr

In [62]:
latlong_ndr.describe(include = 'all')

fid           lat           lng
count  64800.000000  64800.000000  64800.000000
mean   32399.500000      0.000000      0.000000
std    18706.293059     51.961123    103.923449
min        0.000000    -89.500000   -179.500000
25%    16199.750000    -44.750000    -89.750000
50%    32399.500000      0.000000      0.000000
75%    48599.250000     44.750000     89.750000
max    64799.000000     89.500000    179.500000

Here the number of entries is actually higher than the number of entries in the `ndr_table`. This is a bit strange, what does this mean?

Moreover, why do we even need to have different mappings for the different contributions of nature? Couldn't they just use the same mappings between id and coordinates? The answer to this is probably that, as suggested before, the world map resolutions of the nature's contributions are different, with coastal risk having the most detailed data.

# latlong_pollination

In [61]:
latlong_pollination.describe(include = 'all')

fid           lat           lng
count  64800.000000  64800.000000  64800.000000
mean   32399.500000      0.000000      0.000000
std    18706.293059     51.961123    103.923449
min        0.000000    -89.500000   -179.500000
25%    16199.750000    -44.750000    -89.750000
50%    32399.500000      0.000000      0.000000
75%    48599.250000     44.750000     89.750000
max    64799.000000     89.500000    179.500000

Once again the same thing, the number of entries are the same as in the `latlong_ndr` table, 64800, while the number of entries in the `poll_table` is just 9160...



# Actual data

We also take a look at the actual data, just to get a quick feel for the actual figures:

In [15]:
cv_table.head()

fid    UN_cur   UN_ssp1   UN_ssp3   UN_ssp5  pop_cur  pop_ssp1  pop_ssp3  \
0    1  3.282099  3.684031  3.684031  4.135186      0.0       0.0       0.0   
1    2  2.924018  3.282099  3.282099  3.684031      0.0       0.0       0.0   
2    3  2.924018  3.282099  3.282099  3.684031      0.0       0.0       0.0   
3    4  2.924018  3.282099  3.282099  3.684031      0.0       0.0       0.0   
4    5  2.924018  3.282099  3.282099  3.684031      0.0       0.0       0.0   

   pop_ssp5  
0       0.0  
1       0.0  
2       0.0  
3       0.0  
4       0.0

Nothing surprising here! The id:s start from 1 and go up. We see an unmet need for the first 5 geographical regions both currently and for the 3 different future scenarios. We also see that nobody in this region actually gets directly affected by any changes to the natural habitats mitigating coastal risk. This could be because nobody lives there, or simply perhaps because there is no coast in this area.

Let's look at the `ndr_table`.

In [67]:
ndr_table.sample(10)

fid        country         region        UN_cur       UN_ssp1  \
22804  62347     Antarctica        UNKNOWN  0.000000e+00  0.000000e+00   
19313  57821     Antarctica        UNKNOWN  0.000000e+00  0.000000e+00   
19654  58402     Antarctica        UNKNOWN  0.000000e+00  0.000000e+00   
5235   11410         Russia        Eurasia  1.692239e+04  1.816605e+04   
16386  37308     Madagascar         Africa  9.303248e+05  9.337796e+05   
18745  54843     Antarctica        UNKNOWN  0.000000e+00  0.000000e+00   
3152    8676  United States  North America  0.000000e+00  0.000000e+00   
4057    9766         Canada  North America  0.000000e+00  0.000000e+00   
9001   17523     Uzbekistan     North Asia  5.205065e+06  5.186471e+06   
23043  62586     Antarctica        UNKNOWN  0.000000e+00  0.000000e+00   

            UN_ssp3       UN_ssp5      rurpopcur     rurpopssp1  \
22804  0.000000e+00  0.000000e+00       0.000000       0.000000   
19313  0.000000e+00  0.000000e+00       0.000000       0.000000   
19654  0.000000e+00  0.000000e+00       0.000000       0.000000   
5235   1.880769e+04  1.868518e+04     403.656067     249.125153   
16386  1.884855e+06  1.044333e+06  152518.453100  143482.968800   
18745  0.000000e+00  0.000000e+00       0.000000       0.000000   
3152   0.000000e+00  0.000000e+00       0.000000       0.000000   
4057   0.000000e+00  0.000000e+00       0.000000       0.000000   
9001   5.188505e+06  5.191992e+06   88007.234380   85123.585940   
23043  0.000000e+00  0.000000e+00       0.000000       0.000000   

          rurpopssp3    rurpopssp5  
22804       0.000000       0.00000  
19313       0.000000       0.00000  
19654       0.000000       0.00000  
5235      345.910858     268.92337  
16386  357988.906300  141235.34380  
18745       0.000000       0.00000  
3152        0.000000       0.00000  
4057        0.000000       0.00000  
9001   145689.765600   74909.36719  
23043       0.000000       0.00000

We note that the **fid** goes above the maximum number of entries in the table (which was 25243). 

Same for `poll_table`:

In [69]:
poll_table.sample(5)

fid       UN_cur      UN_ssp1      UN_ssp3      UN_ssp5      pop_cur  \
7202  34765     0.000000     0.000000    11.692434     8.400987  440889.3125   
1896  15781  2723.543782  2021.725315  1479.895872  6428.801551       0.0000   
3690  20012   381.369303   252.109262   252.109262   422.644779       0.0000   
8549  42324  2372.489062  2346.716089  6964.512636  2648.081952       0.0000   
6025  29370    25.546455    40.074827   118.007407   116.286427  185547.2969   

         pop_ssp1      pop_ssp3     pop_ssp5  
7202  898507.8750  1.173712e+06  895533.8750  
1896       0.0000  0.000000e+00       0.0000  
3690       0.0000  0.000000e+00       0.0000  
8549  279532.4375  0.000000e+00  295725.9063  
6025  300113.8750  4.032141e+05  302275.2500

# Preprocessing

The next step is to perform the preprocessing (if needed) and save the data in new files. 
This might need to be done in several iteration as we don't know at the moment exactly how we will use the data. One thing that will be good to have done already is to join the lat/lon tables with the data tables.